telemetry/record.go

```diff
@@ -127,9 +128,46 @@ func (record *Record) injectPayloadWithVin() error {
 			return err
 		}
 		message.Vin = record.Vin
+		transformLocations(message)
 		record.PayloadBytes, err = proto.Marshal(message)
 		return err
 	default:
 		return nil
 	}
 }
+
+// transformLocations does a best-effort attempt to convert the Location field to a proper protos.Location
+// type if what we receive is a string that can be parsed. This should make the transition from strings to
+// Locations easier to handle downstream.
+func transformLocations(message *protos.Payload) {
+	for _, datum := range message.Data {
+		if datum.GetKey() == protos.Field_Location && datum.GetValue().GetStringValue() != "" {
```

Comment:
is `datum.GetValue().GetStringValue()` a standard/performant way to check a oneof type?  In the past I've seen this as a `switch`

Reply:
It does a [type assertion](https://github.com/vmallet/fleet-telemetry/blob/e80e50ca3a14cd18c0dd6efb3150c297e2aae714/protos/vehicle_data.pb.go#L627) which is pretty straightforward; here it could be a type switch but with only one case, it's probably no different.

In any case it's done only in the case of a Location field, so once per message at most (well unless we have more locations somehow).